test/integration/hds_integration_test.cc

```diff
@@ -611,15 +674,57 @@ TEST_P(HdsIntegrationTest, TwoEndpointsDifferentClusters) {
 
   // Receive updates until the one we expect arrives
   ASSERT_TRUE(hds_stream_->waitForGrpcMessage(*dispatcher_, response_));
-  while (!(checkEndpointHealthResponse(response_.endpoint_health_response().endpoints_health(0),
+  while (!(checkEndpointHealthResponse(response_.endpoint_health_response()
+                                           .cluster_endpoints_health(0)
+                                           .locality_endpoints_health(0)
+                                           .endpoints_health(0),
                                        envoy::config::core::v3::UNHEALTHY,
                                        host_upstream_->localAddress()) &&
-           checkEndpointHealthResponse(response_.endpoint_health_response().endpoints_health(1),
+           checkEndpointHealthResponse(response_.endpoint_health_response()
+                                           .cluster_endpoints_health(1)
+                                           .locality_endpoints_health(0)
+                                           .endpoints_health(0),
                                        envoy::config::core::v3::HEALTHY,
                                        host2_upstream_->localAddress()))) {
     ASSERT_TRUE(hds_stream_->waitForGrpcMessage(*dispatcher_, response_));
   }
 
+  // Endpoint health response nested layout assertions.
+  ASSERT_EQ(response_.endpoint_health_response().cluster_endpoints_health_size(), 2);
+  ASSERT_EQ(response_.endpoint_health_response().cluster_endpoints_health(0).cluster_name(),
```

Comment:
ASSERT_EQ for repeated field size (line 693) is preferred to avoid accessing out of bound.
Others may use EXPECT_EQ to check field values.